src/core/execute.c

```diff
@@ -872,8 +872,10 @@ static int setup_pam(
 
         STRV_FOREACH(e, *env) {
                 pam_code = pam_putenv(handle, *e);
-                if (pam_code != PAM_SUCCESS)
+                if (pam_code != PAM_SUCCESS) {
+                        e = NULL;
```

Comment:
Indeed, this fixes the issue. Thanks!
But `e=NULL` is tricky. I think we should introduce a new variable (as you suggested: https://github.com/systemd/systemd/pull/3503#discussion_r69868046)